src/librustc_mir/interpret/eval_context.rs

```diff
@@ -95,6 +134,52 @@ pub struct Frame<'mir, 'tcx: 'mir> {
     pub stmt: usize,
 }
 
+impl<'mir, 'tcx: 'mir> Eq for Frame<'mir, 'tcx> {}
+
+impl<'mir, 'tcx: 'mir> PartialEq for Frame<'mir, 'tcx> {
+    fn eq(&self, other: &Self) -> bool {
+        let Frame {
+            mir,
+            instance: _,
+            span: _,
+            return_to_block,
+            return_place,
+            locals,
+            block,
+            stmt,
+        } = self;
+
+        ptr::eq(mir, &other.mir)
```

Comment:
You need to check the `instance` for equality, because it contains the `substs`, which make up the generic arguments of the current frame's function. It's probably not problematic due to the upper frame's `stmt` field, but we should still have this `PartialEq` impl be correct.

This means you also don't have to check the Mir, because the Mir depends directly on the `DefId` inside the `Instance`.

Reply:
Got it, I've read the rustc book, but some of the finer points eluded me.